File: 1768-merge-strings-alternately/1768-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let index = 0;
    let result = "";
    let w1 = word1.length;
    let w2 = word2.length;
    
    while (index < w1 && index < w2){
        result += word1[index];
        result += word2[index];
        index += 1;
    }
    
    if (index < w1){
        result += word1.slice(index);
    }
    
    if (index < w2){
        result += word2.slice(index);
    }
    
    return result;
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        idx = 0
        result = ""
        flag = True
        
        while idx < len(word1) and idx < len(word2):
            result += word1[idx]
            result += word2[idx]
            idx += 1
        
        if idx < len(word1):
            result += word1[idx:]
        
        if idx < len(word2):
            result += word2[idx:]
            
        return result
            